### PR TITLE
Modified build.sh to accomodate static builds.

### PR DESCRIPTION
**Pull request by cosmo83**
#### Description: 

When building cf containers with cf-cli and adding Multiapps plugin, the installation of the same fails because the plugin is not static build but is dependent on libraries, which are not being provided by the container.

> 
> phani@linux-gyok:~/go/src/github.com/cloudfoundry-incubator/multiapps-cli-plugin> ldd build/mta_plugin_linux_amd64 
> 	linux-vdso.so.1 (0x00007fff57996000)
> 	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f3736f2d000)
> 	libc.so.6 => /lib64/libc.so.6 (0x00007f3736b8c000)
> 	/lib64/ld-linux-x86-64.so.2 (0x00007f373714a000)
>

### CODE REVIEW

**Comment by ddonchev** [comment]
Hi @cosmo83,  
We'll think about switching entirely to a static build and for publishing only so-compiled binaries in the cf community cli plugin repo. 
In the mean time, would you like to share the scenario in which you'd be doing mta deployments from an alpine container? 
Would you be using some container pipeline scheduler as concourse, jenkins or some cloud service to automate MTA deployments from such light-weight containers? 
How would the deployed MTA be accessed from within the container e.g. a pre-downloaded file / just built binaries / externally mounted file? 
If the MTA is to be downloaded - would an ability of the 'cf deploy' to accept the mta archive to stdin be useful? e.g. `curl .... | cf deploy - `.
Thank you for the PR!

File: build/changelog [hunk 1]
```diff
@@ -0,0 +1 @@
+* Send mtaId when deployment starts
```
Comment (enchobelezirev) [inline build/changelog]: I think this line does not describe the change. Could you change it?

File: build/version [hunk 2]
```diff
@@ -0,0 +1 @@
+v2.2.1
```
Comment (enchobelezirev) [inline build/version]: This should not be increased unless a new release of the plugin is made.

File: build.sh [hunk 3]
```diff
@@ -12,10 +12,25 @@ function build() {
...
 
+function buildstatic() {
+    local version=$1
+    local platform=$2
+    local arch=$3
+    local plugin_name=$4
+
+    echo calling to build static for $platform $arch
+    CGO_ENABLED=0 GOOS=$platform GOARCH=$arch go build -a -tags netgo \
+        -ldflags "-w -extldflags \"-static\" -X main.Version=${version}" \
```
Comment (enchobelezirev) [inline build.sh]: Just wondering - which binaries which you require are missing in the final produced build?

**Comment by cosmo83** [comment]
> Hi @cosmo83,
> We'll think about switching entirely to a static build and for publishing only so-compiled binaries in the cf community cli plugin repo.
> In the mean time, would you like to share the scenario in which you'd be doing mta deployments from an alpine container?
> Would you be using some container pipeline scheduler as concourse, jenkins or some cloud service to automate MTA deployments from such light-weight containers?
> How would the deployed MTA be accessed from within the container e.g. a pre-downloaded file / just built binaries / externally mounted file?
> If the MTA is to be downloaded - would an ability of the 'cf deploy' to accept the mta archive to stdin be useful? e.g. `curl .... | cf deploy - `.
> Thank you for the PR!

I was actually looking to deploy the same in a alpine container and had issues. The alpine container was mainly serving a concourse build. This was mainly because i was looking to deploy the MTA plugin for a SAP CF build in concourse. 

I have initially started the build with alpine, but because of the issue with MTA plugin and also an issue with MBT (SAP Cloud Build Tool), which was also struggling with alpine, i had to move to a ubuntu container.

File: build.sh (hunk 3 above)
Comment (cosmo83) [reply build.sh]: I was building only for all the platforms. All build were succesful atleast on my system.

**Comment by enchobelezirev** [comment]
Could you squash you commits into one or two and we are ready to merge it ???

**Comment by cosmo83** [comment]
> Could you squash you commits into one or two and we are ready to merge it ???

Done

**Comment by nictas** [comment]
Can you also please sign the individual/corporate CLA? Link is provided by the EasyCLA bot. Sorry to make you jump through this hoop, but it is required of us as well.

**Comment by nictas** [comment]
Thanks @cosmo83 for the contribution! :)

**Pull request merged**